pkg/stack/unwind/compact_unwind_table.go

```diff
@@ -109,20 +121,27 @@ func BuildCompactUnwindTable(fdes frame.FrameDescriptionEntries) (CompactUnwindT
 }
 
 // rowToCompactRow converts an unwind row to a compact row.
-func rowToCompactRow(row *UnwindTableRow) (CompactUnwindTableRow, error) {
+func rowToCompactRow(row *UnwindTableRow, arch elf.Machine) (CompactUnwindTableRow, error) {
 	var cfaType uint8
 	var rbpType uint8
 	var cfaOffset int16
 	var rbpOffset int16
+	var lrOffset int16
 
 	// CFA.
 	//nolint:exhaustive
 	switch row.CFA.Rule {
 	case frame.RuleCFA:
-		if row.CFA.Reg == frame.X86_64FramePointer {
+		// the values for registers are just numerical constants and can overlap(TODO: use a better word)
+		// so checking for machine architecture of the binary is necessary
+		if row.CFA.Reg == frame.X86_64FramePointer && arch == elf.EM_X86_64 {
```

Comment:
For me, moving the arch check to a level above would make it easier. And group the branches per arch rather than inlining here.

Reply:
Totally agree! But how do you feel about putting in a TODO or opening an issue for this now and fixing in a follow-up PR(after initial arm64 DWARF unwinding is merged)? I actually would like to reuse the constant types in which case we maybe can do away with the arch checks entirely.

Reply:
I possible let's do what Kemal suggested as it will make the code easier to understand, but we can always do later on if that's something that you would prefer